packages/xsnap/src/xsnap.js

```diff
@@ -0,0 +1,229 @@
+// @ts-check
+/* eslint no-await-in-loop: ["off"] */
+
+/**
+ * @template T
+ * @typedef {import('./defer').Deferred<T>} Deferred
+ */
+
+import { spawn } from 'child_process';
+import { type as getOsType } from 'os';
+import { defer } from './defer';
+import * as netstring from './netstring';
+import * as node from './node-stream';
+
+const OK = '.'.charCodeAt(0);
+const ERROR = '!'.charCodeAt(0);
+const QUERY = '?'.charCodeAt(0);
+
+const importMetaUrl = `file://${__filename}`;
+
+const osType = getOsType();
+const platform = {
+  Linux: 'lin',
+  Darwin: 'mac',
+  Windows_NT: 'win',
+}[osType];
+
+if (platform === undefined) {
+  throw new Error(`xsnap does not support platform ${osType}`);
+}
+
+const xsnapBin = new URL(
+  `../build/bin/${platform}/release/xsnap`,
+  importMetaUrl,
+).pathname;
+const xsnapDebugBin = new URL(
+  `../build/bin/${platform}/debug/xsnap`,
+  importMetaUrl,
+).pathname;
+
+const encoder = new TextEncoder();
+const decoder = new TextDecoder();
+
+/**
+ * @param {Uint8Array} arg
+ * @returns {Uint8Array}
+ */
+function echoSysCall(arg) {
+  return arg;
+}
+
+/**
+ * @param {Object} options
+ * @param {(request:Uint8Array) => Promise<Uint8Array>} [options.answerSysCall]
+ * @param {string=} [options.name]
+ * @param {boolean=} [options.debug]
+ * @param {string=} [options.snapshot]
+ * @param {'ignore' | 'inherit'} [options.stdout]
+ * @param {'ignore' | 'inherit'} [options.stderr]
+ */
+export function xsnap(options) {
```

Comment:
Let's please have the caller supply `spawn`, following ocap discipline.

See also https://github.com/Agoric/agoric-sdk/issues/2160#issuecomment-753500814 .

Reply:
looks like `getOsType` is ambient too.

Perhaps not a hill I'd die on, but perhaps make `platform` an optional arg and add a "// Note well: AMBIENT AUTHORITY is used for convenience" comment about computing the default.

I also much prefer that module level `const` is only used for things that are design-time const, not things that can change at runtime. So I would prefer that `getOsType` were only called if the caller doesn't supply `platform`.